pkg/kv/k8srestapi/k8srestapi.go

```diff
@@ -0,0 +1,193 @@
+		config, err = rest.InClusterConfig()
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("error creating k8s config: %s", err.Error())
+	}
+
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("error creating k8s client: %s", err.Error())
+	}
+
+	var ownerReference *metav1.OwnerReference
+	ownerReferenceJSON := os.Getenv(EnvK8SOwnerReference)
+	if ownerReferenceJSON != "" {
+		ownerReference = &metav1.OwnerReference{}
+		err := json.Unmarshal([]byte(ownerReferenceJSON), ownerReference)
+		if err != nil {
+			return nil, fmt.Errorf("error unmarhsaling OwnerReference: %s", err.Error())
+		}
+	}
+
+	service = &k8srestapiStore{client, namespace, secret, keyidName, encryptionUrl, decryptionUrl, ownerReference}
+
+	return
+}
+
+func (k *k8srestapiStore) encrypt(plainText []byte) []byte {
+
+	var ep encrypted_payload
+	jsonData := &plaintext_payload{
+		Application:  k.keyidName,
+		PlaintextB64: base64.StdEncoding.EncodeToString(plainText),
+	}
+	jsonValue, _ := json.Marshal(jsonData)
+	request, _ := http.NewRequest("POST", k.encryptionUrl, bytes.NewBuffer(jsonValue))
+	request.Header.Set("Content-Type", "application/json")
+	client := &http.Client{}
+	response, err := client.Do(request)
+	if err != nil {
+		fmt.Printf("The http request failed %s\n", err)
+	} else {
+		cipherTextTemp, _ := ioutil.ReadAll(response.Body)
+		err := json.Unmarshal(cipherTextTemp, &ep)
+		if err != nil {
+			panic(err)
+		}
+	}
+	return ([]byte(ep.EncryptedB64))
+}
+
+func (k *k8srestapiStore) decrypt(cipherText []byte) []byte {
+
+	var pp plaintext_payload
+	jsonData := &encrypted_payload{
+		Application:  k.keyidName,
+		EncryptedB64: string(cipherText),
+	}
+	jsonValue, _ := json.Marshal(jsonData)
+	request, _ := http.NewRequest("POST", k.decryptionUrl, bytes.NewBuffer(jsonValue))
+	request.Header.Set("Content-Type", "application/json")
+	client := &http.Client{}
+	response, err := client.Do(request)
+	if err != nil {
+		fmt.Printf("The http request failed %s\n", err)
+	} else {
+		plainTextTemp, _ := ioutil.ReadAll(response.Body)
+		err := json.Unmarshal(plainTextTemp, &pp)
+		if err != nil {
+			panic(err)
+		}
+	}
+	decoded, err := base64.StdEncoding.DecodeString(pp.PlaintextB64)
+	if err != nil {
+		panic(err)
+	}
+	return decoded
+}
+
+func (k *k8srestapiStore) Set(key string, val []byte) error {
+
+	secret, err := k.cl.CoreV1().Secrets(k.namespace).Get(k.secret, metav1.GetOptions{})
+
+	cipherText := k.encrypt(val)
+	if errors.IsNotFound(err) {
+		secret := &v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: k.namespace,
+				Name:      k.secret,
+			},
+			Data: map[string][]byte{key: cipherText},
+		}
+		if k.ownerReference != nil {
+			secret.ObjectMeta.SetOwnerReferences([]metav1.OwnerReference{*k.ownerReference})
+		}
+		secret, err = k.cl.CoreV1().Secrets(k.namespace).Create(secret)
+
+	} else if err == nil {
+		secret.Data[key] = cipherText
+		secret, err = k.cl.CoreV1().Secrets(k.namespace).Update(secret)
```

Comment:
Please handle this `err`.

Reply:
This function is copied as it is from k8s.go :) 
below will work for this ?
```suggestion
		       } else if err == nil {
                secret.Data[key] = cipherText
                secret, err = k.cl.CoreV1().Secrets(k.namespace).Update(secret)
                //reflect.DeepEqual()
                if err != nil {
                        return fmt.Errorf("error updating secret key '%s' into secret '%s': '%s'", key, k.secret, err.Error())
                }
```

Reply:
Ah sorry, missed the context, looks great!

Reply:
Thanks